stdlib/database/sql/src/main/ballerina/sql/types.bal

```diff
@@ -14,35 +14,71 @@
+#                Default value of this field can be set through the configuration API with the key "connection.auto.commit"
+# + isXA - Whether Connections are used for a distributed transaction.
+#          Default value of this field can be set through the configuration API with the key "xa.enabled"
+# + maximumPoolSize - Maximum size that the pool is allowed to reach, including both idle and in-use connections.
+#                     Default value of this field can be set through the configuration API with the key "max.pool.size"
+# + connectionTimeout - Maximum number of milliseconds that a client will wait for a connection from the pool. Default is 30 seconds.
+#                       Default value of this field can be set through the configuration API with the key "connection.time.out"
+# + idleTimeout - Maximum amount of time that a connection is allowed to sit idle in the pool. Default is 10 minutes.
+#                 Default value of this field can be set through the configuration API with the key "connection.idle.time.out"
+# + minimumIdle - Minimum number of idle connections that pool tries to maintain in the pool. Default is same as maximumPoolSize.
+#                 Default value of this field can be set through the configuration API with the key "connection.min.idle.count"
 # + maxLifetime - Maximum lifetime of a connection in the pool. Default is 30 minutes
+#                 Default value of this field can be set through the configuration API with the key "connection.max.life.time"
 # + validationTimeout - Maximum amount of time that a connection will be tested for aliveness. Default 5 seconds
+#                       Default value of this field can be set through the configuration API with the key "validation.time.out"
 public type PoolOptions record {
-    string connectionInitSql = "";
-    string dataSourceClassName = "";
-    boolean autoCommit = true;
-    boolean isXA = false;
-    int maximumPoolSize = 10;
-    int connectionTimeout = 30000;
-    int idleTimeout = 600000;
-    int minimumIdle = -1;
-    int maxLifetime = 1800000;
-    int validationTimeout = 5000;
+    string connectionInitSql = config:getAsString("connection.init.sql", default = "");
+    string dataSourceClassName = config:getAsString("datasource.class.name", default = "");
+    boolean autoCommit = config:getAsBoolean("connection.auto.commit", default = true);
+    boolean isXA = config:getAsBoolean("xa.enabled", default = false);
+    int maximumPoolSize = config:getAsInt("max.pool.size", default = 10);
+    int connectionTimeout = config:getAsInt("connection.time.out", default = 30000);
+    int idleTimeout =  config:getAsInt("connection.idle.time.out", default = 600000);
+    int minimumIdle = config:getAsInt("connection.min.idle.count", default = 10);
+    int maxLifetime = config:getAsInt("connection.max.life.time", default = 1800000);
+    int validationTimeout = config:getAsInt("validation.time.out", default = 5000);
```

Comment:
IIRC we had a convention to start all conf with "b7a."

Reply:
+1. May be we should add "b7a.database" as the prefix to make it easy to understand where ther property belongs.

Reply:
How about "b7a.sql" prefix as it is short and those are actually belonging to sql package?